builder/dockerfile/dispatchers.go

```diff
@@ -284,52 +437,58 @@ func (b *Builder) getFromImage(shlex *ShellLex, name string) (builder.Image, err
 	}
 	return imageMount.Image(), nil
 }
-
-func processOnBuild(req dispatchRequest) error {
-	dispatchState := req.state
-	// Process ONBUILD triggers if they exist
-	if nTriggers := len(dispatchState.runConfig.OnBuild); nTriggers != 0 {
-		word := "trigger"
-		if nTriggers > 1 {
-			word = "triggers"
-		}
-		fmt.Fprintf(req.builder.Stderr, "# Executing %d build %s...\n", nTriggers, word)
+func (b *Builder) getFromImage(shlex *ShellLex, name string) (builder.Image, error) {
+	name, err := b.getExpandedImageName(shlex, name)
+	if err != nil {
+		return nil, err
 	}
+	return b.getImageOrStage(name)
+}
 
-	// Copy the ONBUILD triggers, and remove them from the config, since the config will be committed.
-	onBuildTriggers := dispatchState.runConfig.OnBuild
-	dispatchState.runConfig.OnBuild = []string{}
-
-	// Reset stdin settings as all build actions run without stdin
-	dispatchState.runConfig.OpenStdin = false
-	dispatchState.runConfig.StdinOnce = false
-
-	// parse the ONBUILD triggers by invoking the parser
-	for _, step := range onBuildTriggers {
-		dockerfile, err := parser.Parse(strings.NewReader(step))
-		if err != nil {
-			return err
-		}
-
-		for _, n := range dockerfile.AST.Children {
-			if err := checkDispatch(n); err != nil {
-				return err
-			}
-
-			upperCasedCmd := strings.ToUpper(n.Value)
-			switch upperCasedCmd {
-			case "ONBUILD":
-				return errors.New("Chaining ONBUILD via `ONBUILD ONBUILD` isn't allowed")
-			case "MAINTAINER", "FROM":
-				return errors.Errorf("%s isn't allowed as an ONBUILD trigger", upperCasedCmd)
-			}
-		}
-
-		if _, err := dispatchFromDockerfile(req.builder, dockerfile, dispatchState); err != nil {
-			return err
-		}
+// TODO: At dispatch time ? Drop ONBUILD support
+
+// func processOnBuild(req parseRequest) error {
+// 	image, err := req.builder.getFromImage(req.shlex, req.args[0])
+// 	if err != nil {
+// 		return err
+// 	}
+// 	onBuilds := image.RunConfig().OnBuild
+// 	// Process ONBUILD triggers if they exist
+// 	// parse the ONBUILD triggers by invoking the parser
+// 	for _, step := range onBuilds {
+// 		dockerfile, err := parser.Parse(strings.NewReader(step))
+// 		if err != nil {
+// 			return err
+// 		}
+
+// 		for _, n := range dockerfile.AST.Children {
```

Comment:
Onbuild will call `command.Parse()` for every item. Then inject the parsed commands into the commands array of currrent stage(or if it simpler it can just call dispatch on them for now).